src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -720,4 +723,117 @@ private static Boolean onlyParseBooleanValue(XContentParser parser) throws IOExc
+            }
+        }
+
+        // Iterate over each rule
+        for (Rule rule : rules) {
+            if (rule == null || rule.getConditions() == null) {
+                // Invalid rule or conditions list is null
+                this.errorMessage = "Suppression Rule Error: A suppression rule or its conditions are not properly defined.";
+                this.issueType = ValidationIssueType.RULE;
+                return;
+            }
+
+            // Iterate over each condition in the rule
+            for (Condition condition : rule.getConditions()) {
+                if (condition == null) {
+                    // Invalid condition
+                    this.errorMessage = "Suppression Rule Error: A condition within a suppression rule is not properly defined.";
+                    this.issueType = ValidationIssueType.RULE;
+                    return;
+                }
+
+                String featureName = condition.getFeatureName();
+
+                // Check if the feature name is null
+                if (featureName == null) {
+                    // Feature name is required
+                    this.errorMessage = "Suppression Rule Error: A condition is missing the feature name.";
+                    this.issueType = ValidationIssueType.RULE;
+                    return;
+                }
+
+                // Check if the feature exists
+                if (!featureEnabledMap.containsKey(featureName)) {
+                    // Feature does not exist
+                    this.errorMessage = "Suppression Rule Error: Feature \""
+                        + featureName
+                        + "\" specified in a suppression rule does not exist.";
+                    this.issueType = ValidationIssueType.RULE;
+                    return;
+                }
+
+                // Check if the feature is enabled
+                if (!featureEnabledMap.get(featureName)) {
+                    // Feature is not enabled
+                    this.errorMessage = "Suppression Rule Error: Feature \""
+                        + featureName
+                        + "\" specified in a suppression rule is not enabled.";
+                    this.issueType = ValidationIssueType.RULE;
+                    return;
+                }
+
+                // other threshold types may not have value operand
+                ThresholdType thresholdType = condition.getThresholdType();
+                if (thresholdType == ThresholdType.ACTUAL_OVER_EXPECTED_MARGIN
+                    || thresholdType == ThresholdType.EXPECTED_OVER_ACTUAL_MARGIN
+                    || thresholdType == ThresholdType.ACTUAL_OVER_EXPECTED_RATIO
+                    || thresholdType == ThresholdType.EXPECTED_OVER_ACTUAL_RATIO) {
+                    // Check if the value is not NaN
+                    double value = condition.getValue();
+                    if (Double.isNaN(value)) {
```

Comment:
btw I will be changing value section to be okay with Null if operator type is ACTUAL_IS_OVER_EXPECTED or EXPECTED_IS_OVER_ACTUAL. Will make sure those aren't passed to RCF and seen as not a rule in that case later on too

Reply:
thanks.